### PR TITLE
chore(api): pkg-upgrade boto3@1.28.57

### PR DESCRIPTION
**Pull request by hamishhossack**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [ ] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Upgrade `boto3` package to the current latest [1.28.57](https://github.com/boto/boto3/releases/tag/1.28.57), include documentation for using import or export within a local environment using `localstack` or similar.

## How did you test this code?

I actually didn't find any good way to test this as a full solution, I guess `pytest` will cover some parts but there could be some potential side-effects where `boto3` is used.

Any guidance here would be much appreciated.

_Related Issue: https://github.com/Flagsmith/flagsmith/issues/2811_

### CODE REVIEW

File: docs/docs/deployment/configuration/importing-and-exporting.md [hunk 1]
```diff
@@ -141,6 +141,13 @@ make sure whichever role you are using to run you container has access to read f
 Alternatively, you can provide the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment variables to refer to an
 IAM user that has access to the S3 bucket.
 
+> Using localstack to achieve local/test exports with s3 can be done using
```
Comment (dabeeeenster) [inline docs/docs/deployment/configuration/importing-and-exporting.md]: Can we add this just as a regular heading - we don't use Quote formatting anywhere in the docs - thanks!

File: docs/docs/deployment/configuration/importing-and-exporting.md [hunk 2]
```diff
@@ -158,3 +165,10 @@ e.g.
 ```bash
 python manage.py import-organisation-from-s3 my-export-bucket exports/organisation-1.json
 ```
+
+> Using localstack to achieve local/test imports with s3 can be done using
```
Comment (dabeeeenster) [inline docs/docs/deployment/configuration/importing-and-exporting.md]: Can we add this just as a regular heading - we don't use Quote formatting anywhere in the docs - thanks!

**Comment by matthewelwell** [comment]
@hamishhossack I think the test pytest coverage should cover us here. I've checked the release notes for the 2 minor versions and I can't see anything that would cause us any problems. Could you resolve the conflicts here and we can approve / merge?

**Comment by matthewelwell** [comment]
@hamishhossack I'm going to close this PR. Please re-open it once you're able to fix the conflicts. Thanks!

**Comment by hamishhossack** [comment]
@matthewelwell Hey, sorry was on holiday!

It seems I don't have the power to re-open this specific PR, seems a bit drastic to create another one... can you re-open?

The rebased has been completed.

**Comment by matthewelwell** [comment]
Erk, looks like Github be doing Github things so I don't think I can reopen it either 😞

image

I think you'll have to create another PR, I'm sorry.

**Comment by hamishhossack** [comment]
re-created the [PR here](https://github.com/Flagsmith/flagsmith/pull/2896):

**Pull request closed without merging**